udf/engine_searches_struct.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TEMP FUNCTION  udf_engine_searches_struct() AS (
+  STRUCT(
+    udf_zeroed_array(12) AS total_searches,
+    udf_zeroed_array(12) AS tagged_searches,
```

Comment:
these should probably be replaced with a function like `udf_12_zeroes` that returns a constant and doesn't use a subquery expression, because a fully denormalized query can contain a limited number of subquery expressions, and because the query optimizer does nice things with constants.